source/base/utilities.cc

```diff
@@ -132,7 +132,7 @@ namespace Utilities
   trim(const std::string &input)
   {
     std::string::size_type left = 0;
-    std::string::size_type right = input.size() - 1;
+    std::string::size_type right = input.size() > 0 ? input.size() - 1 : 0;
```

Comment:
Worth mentioning - my toolset is Visual Studio 2015, it may behave differently using gcc on Linux, or even mingw or intel compiler, or older VS versions on windows. As I wrote - without this fix, step-33 (and most probably other tests as well) fails using VS 2015.